src/compiler/commandLineParser.ts

```diff
@@ -3042,6 +3082,98 @@ function parseJsonConfigFileContentWorker(
     }
 }
 
+/** @internal */
+export function handleWatchOptionsConfigDirTemplateSubstitution(

```

Comment:
Can we eliminate the `createCopyOnSubstitute` variable here, and have these *always* return a new `options` object? And then at callsites where they want to update the original object, they can explicitly merge the new object into the old one? It seems like every callsite sets this to `true` except one, so it feels like it'd be a bit cleaner that way.